cpp/csp/python/PyAdapterManager.cpp

```diff
@@ -68,6 +68,25 @@ class PyAdapterManager : public AdapterManager
 static PyObject * PyAdapterManager_PyObject_starttime( PyAdapterManager_PyObject * self ) { return toPython( self -> manager -> starttime() ); }
 static PyObject * PyAdapterManager_PyObject_endtime( PyAdapterManager_PyObject * self )   { return toPython( self -> manager -> endtime() ); }
 
+static PyObject * PyAdapterManager_PyObject_shutdown_engine( PyAdapterManager_PyObject * self, PyObject * args, PyObject * kwargs )
+{
+    CSP_BEGIN_METHOD;
+
+    PyObject * pyException;
+    if( !PyArg_ParseTuple( args, "O", &pyException ) )
+        return NULL;
+    
+    if( !PyExceptionInstance_Check( pyException ) )
+    {
+        std::string desc = "Expected Exception object as argument for shutdown_engine: got " + std::string( Py_TYPE( pyException ) -> tp_name );
```

Comment:
What do you guys think about allowing passing a string and just passing that through as is ( Other types can do this, though maybe we should convert the object to a string instead of just logging its typename ) )

Reply:
Users can always pass `Exception('msg')` if they don't have an exception to pass-along to the method. I don't think it's worth it to complicate the API by allowing both strings and typed exceptions.

Reply:
Sure, thats fair.  We should still make the error msg that the user gets back clearer ( basically just replace streaming the type as the PyObject_Str of the object )